cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableService.java

```diff
@@ -0,0 +1,449 @@
+      .setLocation(streamLocation)
+      .setTableComment("CDAP Stream")
+      .buildWithStorageHandler(Constants.Explore.STREAM_STORAGE_HANDLER_CLASS, serdeProperties);
+
+    LOG.debug("Running create statement for stream {}", streamName);
+
+    return exploreService.execute(streamID.getNamespace(), createStatement);
+  }
+
+  /**
+   * Disable exploration on the given stream by dropping the Hive table for the stream.
+   *
+   * @param streamID the ID of the stream to disable
+   * @return the query handle for disabling the stream
+   * @throws ExploreException if there was an exception dropping the table
+   * @throws SQLException if there was a problem with the drop table statement
+   */
+  public QueryHandle disableStream(Id.Stream streamID) throws ExploreException, SQLException {
+    LOG.debug("Disabling explore for stream {}", streamID);
+    String deleteStatement = generateDeleteStatement(getStreamTableName(streamID));
+    return exploreService.execute(streamID.getNamespace(), deleteStatement);
+  }
+
+  /**
+   * Enable ad-hoc exploration on the given dataset by creating a corresponding Hive table. If exploration has
+   * already been enabled on the dataset, this will be a no-op. Assumes the dataset actually exists.
+   *
+   * @param datasetID the ID of the dataset to enable
+   * @param spec the specification for the dataset to enable
+   * @return query handle for creating the Hive table for the dataset
+   * @throws IllegalArgumentException if some required dataset property like schema is not set
+   * @throws UnsupportedTypeException if the schema of the dataset is not compatible with Hive
+   * @throws ExploreException if there was an exception submitting the create table statement
+   * @throws SQLException if there was a problem with the create table statement
+   * @throws DatasetNotFoundException if the dataset had to be instantiated, but could not be found
+   */
+  public QueryHandle enableDataset(Id.DatasetInstance datasetID, DatasetSpecification spec)
+    throws IllegalArgumentException, ExploreException, SQLException,
+    UnsupportedTypeException, DatasetNotFoundException {
+
+    String datasetName = datasetID.getId();
+    Map<String, String> serdeProperties = ImmutableMap.of(
+      Constants.Explore.DATASET_NAME, datasetName,
+      Constants.Explore.DATASET_NAMESPACE, datasetID.getNamespaceId());
+    String createStatement = null;
+
+    // some datasets cannot be instantiated here. For example, ObjectMappedTable is often parameterized with a type
+    // that is only available in a program context and not available here in the system context.
+    // explore should only have logic related to exploration and not dataset logic.
+    // TODO: refactor exploration (CDAP-1573)
+    String datasetType = spec.getType();
+    // special casing here... but we really should clean this up
+    // there are two ways to refer to each dataset type...
+    if (ObjectMappedTableModule.FULL_NAME.equals(datasetType) ||
```

Comment:
don't have a strong opinion, but leaning towards keeping as is, since it's only done here and is just an or.